internal/storage/v1/cassandra/spanstore/writer.go

```diff
@@ -190,24 +189,24 @@ func (s *SpanWriter) writeIndexes(span *model.Span, ds *dbmodel.Span) error {
 		}
 	}
 
-	if span.Flags.IsFirehoseEnabled() {
+	if dbmodel.IsFirehoseEnabled(ds) {
```

Comment:
this logic can be dropped altogether, there is no "firehose" bit in OTLP trace state flags.